src/System.Net.Http/src/System/Net/Http/Unix/CurlTransportContext.cs

```diff
@@ -2,36 +2,19 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
-using System;
 using System.Security.Authentication.ExtendedProtection;
+using System.Threading;
 
 namespace System.Net.Http
 {
-    internal class CurlTransportContext : TransportContext
+    internal sealed class CurlTransportContext : TransportContext
     {
-        private CurlChannelBinding _channelBinding = new CurlChannelBinding();
+        private CurlChannelBinding _channelBinding;
 
-        internal CurlChannelBinding CurlChannelBinding
-        {
-            get
-            {
-                return _channelBinding;
-            }
-        }
+        internal CurlChannelBinding CurlChannelBinding => LazyInitializer.EnsureInitialized(ref _channelBinding);
```

Comment:
I'm not used to seeing this new (C# 5 or 6) syntax.  So,just so I understand this line of code:

* This is defining a new property called "CurlChannelBinding" that has a getter only.
* The property is returning an object of type "CurlChannelBinding".
* The .EnsureInitialized() method is making sure to init (only once) the private field _channelBinding using the default constructor of the "CurlChannelBinding" class?